internal/db/read_writer.go

```diff
@@ -70,6 +70,9 @@ type Reader interface {
 
 	// ScanRows will scan sql rows into the interface provided
 	ScanRows(ctx context.Context, rows *sql.Rows, result any) error
+
+	// Now returns the current transaction timestamp.
+	Now(ctx context.Context) (time.Time, error)
```

Comment:
This logic is imported from #3764 to support setting the timestamp based on the timestamp of the transaction used to list deleted IDs.